### PR TITLE
Feature 11574 environment samples on dashboard map

### LINKED ISSUE #11574: Add environment samples to sample status map [2]
### Feature Description

This issue is part of the EPIC #11533 on the environmental surveillance module. Display environment samples in the status map on the dashboard for users with access to them.

### Acceptance Criteria

- [x] A checkbox "Show environment samples" has been added to the "Layers" dropdown on the sample dashboard; only visible if the user has view access to environment samples
- [x] A symbol for environment samples has been introduced and added to the "Map key" dropdown; use a distinguishable shade of green as its color
- [x] Environment samples that are within the user's jurisdiction (or owned) are shown on the map if the filter is enabled
- [x] The samples are shown based on GPS coordinates; if a sample has GPS coordinates, use those, otherwise fall back to the environment
- [x] The info hint next to the "Sample Status Map" heading has been extended as follows for users with access to environment samples: "Samples are shown using the GPS coordinates of the person's home address. Environment samples are shown using the GPS coordinates of those samples or, if not available, their associated environment."

### PR DESCRIPTION
**Pull request by leventegal-she**
Fixes #11574

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12581

**Review by sergiupacurariu — CHANGES_REQUESTED** [review]

File: sormas-backend/src/main/java/de/symeda/sormas/backend/dashboard/sample/SampleDashboardService.java [hunk 1]
```diff
@@ -207,63 +287,45 @@ public Long countSamplesForMap(SampleDashboardCriteria criteria, Set<SampleAssoc
 		return QueryHelper.getSingleResult(em, cq);
 	}
 
-	public List<MapSampleDto> getSamplesForMap(SampleDashboardCriteria criteria, Set<SampleAssociationType> associationTypes) {
-		if (associationTypes.isEmpty()) {
-			return Collections.emptyList();
-		}
+	public Long countEnvironmentSamplesForMap(SampleDashboardCriteria criteria) {
```
Comment (sergiupacurariu) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/dashboard/sample/SampleDashboardService.java:290]: same like for "getEnvironmentalSamplesForMap" method

**Review by sergiupacurariu — CHANGES_REQUESTED** [review]

File: sormas-api/src/main/java/de/symeda/sormas/api/dashboard/sample/MapSampleDto.java [hunk 2]
```diff
@@ -28,6 +28,11 @@ public class MapSampleDto implements Serializable {
 
 	private SampleAssociationType associationType;
 
+	public MapSampleDto(Double longitude, Double latitude, Double associatedEntityLongitude, Double associatedEntityLatitude) {
```
[the remark this answers is not in the document]
Comment (leventegal-she) [reply sormas-api/src/main/java/de/symeda/sormas/api/dashboard/sample/MapSampleDto.java:31]: it is used by the JPA query of getting the samples for the map

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12581

**Review by sergiupacurariu — APPROVED** [review]

**Pull request merged**